eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -729,6 +730,14 @@ lookupImpRootTxOut = do
     Nothing -> error "Root txId no longer points to an existing unspent output"
     Just rootTxOut -> pure (impRootTxIn, rootTxOut)
 
+impRootKh ::
+  forall era. (EraTxOut era, HasCallStack) => ImpTestM era (KeyHash 'Payment (EraCrypto era))
+impRootKh = do
+  rootTxOut <- snd <$> lookupImpRootTxOut
+  pure $ case rootTxOut ^. addrTxOutL @era of
+    Addr _ (KeyHashObj kh) _ -> kh
+    _ -> error "Missing expected root key hash"
+
```

Comment:
I think this approach is little too hacky. We should not use the root tx for more than one purpose, especially if it can be avoided
```suggestion
```